lib/node_modules/npm/package.json

```diff
@@ -0,0 +1,4 @@
+{
+  "name": "npm",
+  "description": "a package manager for JavaScript; this package.json file helps IntelliJ/WebStorm to recognize the npm directory as package"
```

Comment:
I've confirmed that webstorm doesn't actually need any of this (strictly) and works as long as `bin/npm-cli.js` exists in the folder that's configured as the "package manager". I'm not against including this file, but if so, we should actually use it by declaring the main/bin file appropriately.

Reply:
`package.json` file was added as WebStorm checks its presence. It can be empty. Otherwise, `lib/node_modules/npm` won't be considered as a package.

Reply:
Strange, it doesn't work for me without package.json (possibly empty).

Reply:
The workaround as documented works for me with webstorm 2018.3. https://github.com/nodenv/nodenv/wiki/FAQ#workaround

The workaround just adds `bin/npm-cli.js` to the nodenv root directory, and requires user to point webstorm to the nodenv root directory as the package manager. That was all that was necessary for it to work for me. No lib/node_modules, no package.json, etc. Just bin/npm-cli.js in the directory configured as the package manager.

Reply:
I have a custom location for $NODENV_ROOT, so there is no package.json in mine:

```
$ cd `nodenv root`
$ pwd
/usr/local/var/nodenv
$ ls
bin/              default-packages  nodenv.d/         shims/            versions/
cache/            etc/              plugins/          sources/
$ ls bin/
npm-cli.js
```

> The workaround with lib/node_modules/npm/{package.json, bin/npm-cli.js} won't require users to specify "Package manager" field manually, it will be inferred automatically if "Node interpreter" points to $(nodenv root)/shims/node.

That's good to know but it would be a "weird" install location for nodenv users. nodenv plugins live in `$(nodenv root)/plugins`. My intention for the shim plugin is to be installable as a global npm package _or_ as a nodenv plugin. But both installations would require manually specifying its location via webstorm package manager, since neither approach would end up in $(nodenv root)/lib/node_modules/...

Reply:
Right, I forget that an absolute path to npm package is specified in UI. Existence of `package.json` is checked only when inferring npm package automatically by node interpreter location or npm executable. Manually specified npm packages are not checked for `package.json`.